matter_server/server/device_controller.py

```diff
@@ -662,6 +670,69 @@ def resubscribe() -> None:
             5, resubscribe
         )
 
+    @api_command(APICommand.PING_NODE)
+    async def ping_node(self, node_id: int) -> NodePingResult:
+        """Ping node on the currently known IP-adress(es)."""
+        result: NodePingResult = {}
+        # the node's ip addresses are stored in the GeneralDiagnostics cluster
+        attribute = Clusters.GeneralDiagnostics.Attributes.NetworkInterfaces
+        attr_path = f"0/{attribute.cluster_id}/{attribute.attribute_id}"
+        node = self._nodes.get(node_id)
+        if node is None:
+            raise NodeNotExists(
+                f"Node {node_id} does not exist or is not yet interviewed"
+            )
+        if node.available:
+            # try to refresh the GeneralDiagnostics.NetworkInterface attribute
+            # so we have the most accurate information before pinging
+            try:
+                await self.read_attribute(node_id, attr_path)
+            except (NodeNotResolving, ChipStackError) as err:
+                LOGGER.exception(err)
+
+        battery_powered = (
+            node.attributes.get("0/53/1", 0)
```

Comment:
I guess the idea is to get the cached value here? It is kinda annoying that we have to use the numerical representation here :thinking:

Reply:
yeah, this is a fast key lookup, otherwise I need to look it up from the class+attribute

Reply:
I'll put it in a constant

Reply:
created a constant